### DISCUSSION TITLE
Setting thermal parameters

### DISCUSSION THREAD

**Discussion by ahsadeghi** [pybamm-team/PyBaMM#3068] [Q&A] [answered]
Hello,
It is said in the [examples](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/models/thermal-models.ipynb) that the heat generation is due to the three terms and I can see the three terms in [base_thermal](https://github.com/pybamm-team/PyBaMM/blob/bf8581174460005f4d4ecb1a0b56cbf665cd59ce/pybamm/models/submodels/thermal/base_thermal.py). But I cannot figure out how the Arrhenius equations are coded in this module, and how can I change the Arrhenius parameters such as activation energy and pre-exponential factor. I wonder if you could please explain about this.

Regards,
Hosein

**Comment by ben-kenney** [comment]
You can take a look through the parameter sets and you should see the dU/dT curves for the positive/negative electrodes. For example in [Mohtat2020](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/input/parameters/lithium_ion/Mohtat2020.py) you'll find the entropy change for [graphite ](https://github.com/pybamm-team/PyBaMM/blob/bf8581174460005f4d4ecb1a0b56cbf665cd59ce/pybamm/input/parameters/lithium_ion/Mohtat2020.py#L93) and [NMC](https://github.com/pybamm-team/PyBaMM/blob/bf8581174460005f4d4ecb1a0b56cbf665cd59ce/pybamm/input/parameters/lithium_ion/Mohtat2020.py#L215).

The activation energies appear to be baked into functions, [like this for NMC diffusivity](https://github.com/pybamm-team/PyBaMM/blob/bf8581174460005f4d4ecb1a0b56cbf665cd59ce/pybamm/input/parameters/lithium_ion/Mohtat2020.py#L123-L149):

```
def NMC_diffusivity_PeymanMPM(sto, T):
    D_ref = 8 * 10 ** (-15)
    E_D_s = 18550
    arrhenius = pybamm.exp(E_D_s / pybamm.constants.R * (1 / 298.15 - 1 / T))
    return D_ref * arrhenius
```
It seems to be possible to write your own function where you should be able to specify the activation energy...something I'll also need to do but haven't gotten there yet!

**Comment by ahsadeghi** [comment]
Dear @brosaplanella 
Regarding this question, I need to (for instance) modify `E_r` and `E_D_s` in [ORegan2022.py](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/input/parameters/lithium_ion/ORegan2022.py). I am not sure if I can do it through the input file or do I need to compile the code with the modified values. I wonder if you could please explain this. Thank you.

Hosein

**Reply by brosaplanella** [reply, marked as answer]
You can just copy ORegan2022.py into a new file locally, edit the parameters, and then in your main code import the `get_parameter_values` function from that file.

**Reply by ben-kenney** [reply]
I'm interested in a variation of this: The answer above will create a new function that has a hard coded activation energy but what if we needed to treat the activation energy as a fitting parameter?

Any tips on how to modify the function above `NMC_diffusivity_PeymanMPM(sto, T)` so that the activation energy can be easily updated for a fitting process?

**Reply by brosaplanella** [reply]
Just define it as a parameter:
```python3
E_D_s = pybamm.Parameter("Activation energy of positive electrode diffusivity [J.mol-1]")
```
Then you can set the value by adding it to the dictionary with all the other parameter values.
